module/friendship/adapter/postgres/repository/subscription_test.go

```diff
@@ -0,0 +1,390 @@
+package repository
+
+import (
+	"context"
+	"errors"
+	"log"
+	"testing"
+
+	"github.com/phantranhieunhan/s3-assignment/common"
+	"github.com/phantranhieunhan/s3-assignment/common/adapter/postgres"
+	"github.com/phantranhieunhan/s3-assignment/module/friendship/adapter/postgres/model"
+	"github.com/phantranhieunhan/s3-assignment/module/friendship/domain"
+	"github.com/phantranhieunhan/s3-assignment/pkg/config"
+	"github.com/phantranhieunhan/s3-assignment/pkg/util"
+	"github.com/stretchr/testify/assert"
+	"github.com/volatiletech/sqlboiler/v4/boil"
+)
+
+type Suite struct {
+	db postgres.Database
+}
+
+func NewSuite(ctx context.Context) Suite {
+	err := config.ReadConfig()
+
+	if err != nil {
+		log.Fatal(err)
+	}
+	db := postgres.NewDatabase()
+	return Suite{db: db}
+}
+
+type SubscriptionTestCase struct {
+	name                     string
+	modifySub                domain.Subscription
+	isExisted                bool
+	isFounded                bool
+	mentionedEmails          []string
+	blockedEmails            []string
+	isInvalidMentionedEmails bool
+	result                   []string
+	err                      error
+}
+
+func TestSubscription_Create(t *testing.T) {
+	ctx := context.Background()
+	suite := NewSuite(ctx)
+	repo := NewSubscriptionRepository(suite.db)
+
+	errDB := errors.New("some error from db")
+
+	cs := []SubscriptionTestCase{
+		{
+			name: "successful",
+		},
+		{
+			name: "fail by reference user id",
+			modifySub: domain.Subscription{
+				UserID: "user-id",
+			},
+			err: common.ErrDB(errDB),
+		},
+		{
+			name: "fail by reference subscriber id",
+			modifySub: domain.Subscription{
+				SubscriberID: "subscriber-id",
+			},
+			err: common.ErrDB(errDB),
+		},
+	}
+	for _, tc := range cs {
+		t.Run(tc.name, func(t *testing.T) {
+			sub := domain.Subscription{
+				UserID:       util.GenUUID(),
+				SubscriberID: util.GenUUID(),
+				Status:       domain.SubscriptionStatusSubscribed,
+			}
+			suite.prepareSubscription(t, ctx, sub)
+			createdSub := sub
+			if tc.err != nil {
+				if tc.modifySub.SubscriberID != "" {
+					createdSub.SubscriberID = "subscriber-id"
+				} else if tc.modifySub.UserID != "" {
+					createdSub.UserID = "user-id"
+				}
+			}
+			var err error
+			sub.Id, err = repo.Create(ctx, createdSub)
+			if tc.err == nil {
+				assert.NoError(t, err)
+				assert.NotEmpty(t, sub.Id)
+			} else {
+				assert.Error(t, err)
+				assert.Empty(t, sub.Id)
+			}
+
+			suite.rollbackSubscription(t, ctx, sub, []string{sub.Id})
```

Comment:
Next time, you can start a transaction and use its rollback mechanism. But this is fine, I like your idea